### PR TITLE
Feature/token utility events

### PR DESCRIPTION
**Pull request by MRamanenkau**
## Description

[Link](https://github.com/Cerebellum-Network/network-relayer/pull/180) to Network Relayer companion PR

## Types of Changes
Please select the branch type you are merging and fill in the relevant template.

- [ ] Hotfix
- [ ] Release
- [x] Fix or Feature

### Types of Changes

- [ ] Tech Debt (Code improvements)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Dependency upgrade (A change in substrate or any 3rd party crate version)

### Migrations and Hooks

- [ ] This change requires a runtime migration.
- [ ] Modifies `on_initialize`
- [ ] Modifies `on_finalize`

### Checklist for Fix or Feature

- [x] Change has been tested locally.
- [x] Change adds / updates tests if applicable.
- [x] Changelog doc updated.

## Checklist for Hotfix

- [ ] Change has been deployed to Testnet.
- [ ] Change has been tested in Testnet.
- [ ] Changelog has been updated.
- [ ] Crate version has been updated.
- [ ] Spec version has been updated.
- [ ] Transaction version has been updated if required.
- [ ] Pull Request to `dev` has been created.
- [ ] Pull Request to `staging` has been created.

## Checklist for Release

- [ ] Change has been deployed to Devnet.
- [ ] Change has been tested in Devnet.
- [ ] Change has been deployed to Qanet.
- [ ] Change has been tested in Qanet.
- [ ] Change has been deployed to Testnet.
- [ ] Change has been tested in Testnet.
- [ ] Changelog has been updated.
- [ ] Crate version has been updated.
- [ ] Spec version has been updated.
- [ ] Transaction version has been updated if required.

### CODE REVIEW

File: pallets/ddc-payouts/src/lib.rs [hunk 1]
```diff
@@ -198,14 +198,22 @@ pub mod pallet {
 			cluster_id: ClusterId,
 			era: DdcEra,
 		},
-		Rewarded {
+		ProvidersRewarded {
```
Comment (aie0) [inline pallets/ddc-payouts/src/lib.rs:201]: don't change the names - these events are already captured by DAC

File: pallets/ddc-payouts/src/lib.rs [hunk 2]
```diff
@@ -198,14 +198,22 @@ pub mod pallet {
...
 			batch_index: BatchIndex,
+			stored_bytes: u64,
+			transferred_bytes: u64,
+			number_of_puts: u64,
+			number_of_gets: u64,
 			node_provider_id: T::AccountId,
 			rewarded: u128,
 			expected_to_reward: u128,
 		},
+		ValidatorsRewarded {
```
Comment (aie0) [inline pallets/ddc-payouts/src/lib.rs:213]: add      cluster_id: ClusterId,
			era: DdcEra,

**Review by aie0 — CHANGES_REQUESTED** [review]

File: pallets/ddc-payouts/src/lib.rs [hunk 3]
```diff
@@ -785,10 +793,14 @@ pub mod pallet {
 						.ok_or(Error::<T>::ArithmeticOverflow)?;
 				}
 
-				Self::deposit_event(Event::<T>::Rewarded {
+				Self::deposit_event(Event::<T>::ProvidersRewarded {
```
Comment (aie0) [inline pallets/ddc-payouts/src/lib.rs:796]: @yahortsaryk for the new events, do we need to update the go-SDK?

File: pallets/ddc-payouts/src/lib.rs (hunk 1 above)
Comment (yahortsaryk) [reply pallets/ddc-payouts/src/lib.rs:201]: I think the naming doesn't play a significant role here. The index, the order of fields, and the types are what matters for successful event parsing.

File: pallets/ddc-payouts/src/lib.rs (hunk 2 above)
Comment (MRamanenkau) [reply pallets/ddc-payouts/src/lib.rs:213]: Updated

File: pallets/ddc-payouts/src/lib.rs (hunk 1 above)
Comment (MRamanenkau) [reply pallets/ddc-payouts/src/lib.rs]: Anyway, returned as it was

File: pallets/ddc-payouts/src/lib.rs (hunk 3 above)
Comment (MRamanenkau) [reply pallets/ddc-payouts/src/lib.rs]: I'll update the `network-relayer` and make sure @khssnv is notified

File: pallets/ddc-payouts/src/lib.rs [hunk 4]
```diff
@@ -947,6 +961,11 @@ pub mod pallet {
 				amount_to_deduct.saturated_into::<BalanceOf<T>>(),
 				ExistenceRequirement::AllowDeath,
 			)?;
+
+			pallet::Pallet::deposit_event(Event::<T>::ValidatorsRewarded {
+				validator_id: staker_id.clone(),
+				amount: amount_to_deduct,
```
[the remark this answers is not in the document]
Comment (MRamanenkau) [reply pallets/ddc-payouts/src/lib.rs:971]: Sure, updated

File: pallets/ddc-payouts/src/lib.rs [hunk 5]
```diff
@@ -947,6 +963,13 @@ pub mod pallet {
 				amount_to_deduct.saturated_into::<BalanceOf<T>>(),
 				ExistenceRequirement::AllowDeath,
 			)?;
+
+			pallet::Pallet::deposit_event(Event::<T>::ValidatorsRewarded {
```
Comment (aie0) [inline pallets/ddc-payouts/src/lib.rs:967]: rename ValidatorsRewarded to ValidatorRewarded cause it's only 1 validator

File: pallets/ddc-payouts/src/lib.rs [hunk 6]
```diff
@@ -202,10 +202,20 @@ pub mod pallet {
 			cluster_id: ClusterId,
```
Comment (aie0) [inline pallets/ddc-payouts/src/lib.rs:202]: it makes sense to rename the field. I've checkde with DAC and it should be fine. as we have several rewards, it makes sense to name specificly

please rename to ProviderRewarded

**Review by aie0 — CHANGES_REQUESTED** [review]

File: pallets/ddc-payouts/src/lib.rs (hunk 6 above)
Comment (MRamanenkau) [reply pallets/ddc-payouts/src/lib.rs:202]: Updated

File: pallets/ddc-payouts/src/lib.rs (hunk 5 above)
Comment (MRamanenkau) [reply pallets/ddc-payouts/src/lib.rs]: Updated

**Review by yahortsaryk — APPROVED** [review]

**Review by aie0 — APPROVED** [review]

**Pull request merged**